### ISSUE TITLE
Fix available/away state

### ISSUE THREAD

**Issue by webratte** [kontalk/androidclient#637]
I don't know if it's possible.

My idea is:
-drop "online".
-show "activ" only if the app is open not when push message is arrived. (I can see the second checkmark to know the app is connected. And if I don't send a me sage I don't need to know if the app connected) 
-"last seen time"- the time when the app is closed last time (that's the time when the user is last seen)

IMO Users not interessted in "App connected".
They are interested in activity of the user.

**Comment by daniele-athome** [comment]
Point 2 will probably come on next release or by 3.2 (the current method in 3.1.2 was a start).
Point 3 is... complicated.

Currently, Tigase registers the "last activity time" as the time when the user last logged out of the server (that is, disconnected).

Little digression: connecting to the server is expensive. Very expensive. In fact, in most situations it's more convenient (to both the server and the client) to keep the connection alive and send a ping from time to time.

Disconnecting is what triggers an update of the last activity time (which is what is shown when you read "last seen at"). It should (hopefully) happen rarely. Imagine that a timestamp (e.g. a database update, read: expensive) is updated everytime the app closes - in XMPP terms, it means it should go inactive.

Another issue to consider is the way Tigase handles this. It's very different because it follows the XMPP standard and it would require more customization.
Something from XEP-0256 http://xmpp.org/extensions/xep-0256.html could be used for this, but again Tigase has no support for it (that i know of, though I'm not 100% sure).

TL; DR: it needs some work, but it's not impossible :-)

**Comment by webratte** [comment]
Thank you. Your post is much to technical for me. But I'm sure you know what you do ;-) 

If you can realize point 2, I will be glad.
The other points are not really importent I think.

**Comment by daniele-athome** [comment]
> Note to myself

CSI is currently enabled 30 seconds after last release of message center or when the screen is turned off. At the same time, an away presence is broadcasted.
When a message is received, the message center sets a hold on itself to store and process the message correctly. So an an available presence is broadcasted.

Hold/release mechanism should be actually used only for keeping the service connected to the server.
CSI is activated some time after last released (also away status is broadcasted), however, when calling `hold()` again, it shouldn't deactivate CSI and send an available presence.
